src/composer.ts

```diff
@@ -276,60 +333,91 @@ export class Composer<TContext extends Context>
   static optional<TContext extends Context>(
     predicate: Predicate<TContext> | AsyncPredicate<TContext>,
     ...fns: NonemptyReadonlyArray<Middleware<TContext>>
-  ) {
+  ): Middleware.Fn<TContext> {
     return Composer.branch(
       predicate,
       Composer.compose(fns),
       Composer.passThru()
     )
   }
 
-  static filter<TContext extends Context>(predicate: Predicate<TContext>) {
+  static filter<TContext extends Context>(
+    predicate: Predicate<TContext>
+  ): Middleware.Fn<TContext> {
     return Composer.branch(predicate, Composer.passThru(), anoop)
   }
 
-  static drop<TContext extends Context>(predicate: Predicate<TContext>) {
+  static drop<TContext extends Context>(
+    predicate: Predicate<TContext>
+  ): Middleware.Fn<TContext> {
     return Composer.branch(predicate, anoop, Composer.passThru())
   }
 
   static dispatch<
     TContext extends Context,
     Handlers extends Record<string | number | symbol, Middleware<TContext>>
-  >(routeFn: (ctx: TContext) => keyof Handlers, handlers: Handlers) {
+  >(
+    routeFn: keyof Handlers | ((ctx: TContext) => keyof Handlers),
```

Comment:
I wanna drop this behavior, not document it. Users can easily enough `.use(handlers[handlerKey])`.

Reply:
Should we just remove it entirely then? After all, there's no better point in time than a new major version to drop legacy code.